cpr/session.cpp

```diff
@@ -272,6 +272,14 @@ void Session::SetUrl(const Url& url) {
     url_ = url;
 }
 
+void Session::SetResolve(const Resolve& resolve) {
```

Comment:
In my eyes `Session::SetResolve(...)` should override all previous existing resolves. In this case it should clear the current list first and then it should append the new values.

Also it might be useful to pass multiple `cpr::Resolve` objects to this function. Therefore we should also provide something like this: `Session::SetResolve(const std::vector<Resolve>& resolves)`. The same goes then obviously for the `SetOption(...)` call.